--- a/glib/gatomic.h
+++ b/glib/gatomic.h
@@ -167,28 +167,28 @@
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     (void) (0 ? (gpointer) *(atomic) : 0);                                   \
     (void) (0 ? (val) ^ (val) : 0);                                          \
-    (gssize) __sync_fetch_and_add ((atomic), (val));                         \
+    (gssize) __sync_fetch_and_add ((volatile gssize *) (atomic), (val));     \
   }))
 #define g_atomic_pointer_and(atomic, val) \
   (G_GNUC_EXTENSION ({                                                          \
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     (void) (0 ? (gpointer) *(atomic) : 0);                                   \
     (void) (0 ? (val) ^ (val) : 0);                                          \
-    (gsize) __sync_fetch_and_and ((atomic), (val));                          \
+    (gsize) __sync_fetch_and_and ((volatile gsize *) (atomic), (val));       \
   }))
 #define g_atomic_pointer_or(atomic, val) \
   (G_GNUC_EXTENSION ({                                                          \
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     (void) (0 ? (gpointer) *(atomic) : 0);                                   \
     (void) (0 ? (val) ^ (val) : 0);                                          \
-    (gsize) __sync_fetch_and_or ((atomic), (val));                           \
+    (gsize) __sync_fetch_and_or ((volatile gsize *) (atomic), (val));        \
   }))
 #define g_atomic_pointer_xor(atomic, val) \
   (G_GNUC_EXTENSION ({                                                          \
     G_STATIC_ASSERT (sizeof *(atomic) == sizeof (gpointer));                 \
     (void) (0 ? (gpointer) *(atomic) : 0);                                   \
     (void) (0 ? (val) ^ (val) : 0);                                          \
-    (gsize) __sync_fetch_and_xor ((atomic), (val));                          \
+    (gsize) __sync_fetch_and_xor ((volatile gsize *) (atomic), (val));       \
   }))
 
 #else /* defined(G_ATOMIC_LOCK_FREE) && defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) */
